platform_plugin_aspects/xblock.py

```diff
@@ -0,0 +1,175 @@
+"""XBlock to embed Superset dashboards in Open edX."""
+
+from __future__ import annotations
+
+import logging
+
+import pkg_resources
+from django.utils import translation
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import List, Scope, String
+from xblock.utils.resources import ResourceLoader
+from xblock.utils.studio_editable import StudioEditableXBlockMixin
+
+from .utils import _, generate_superset_context
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+@XBlock.needs("user")
+@XBlock.needs("i18n")
+class SupersetXBlock(StudioEditableXBlockMixin, XBlock):
```

Comment:
Do we want the XBlock to be called SupersetXBlock, or AspectsXBlock?

Reply:
I would prefer `SupersetXBlock` as it doesn't have any integration with Aspects except the default URLs and credentials.

Eventually, this can be refactored/extended a bit to support the specific learner report or a report specific to the current component via an Aside or something similar but it's not the roadmap for now.